lib/Optimizer/Transforms/QuakeSynthesizer.cpp

```diff
@@ -115,6 +115,103 @@ static bool hasInitStateUse(BlockArgument argument) {
   return false;
 }
 
+template <typename T>
+Value createGlobalArray(OpBuilder &builder, ModuleOp module, unsigned &counter,
+                        BlockArgument argument, Type arrTy,
+                        std::vector<T> vec) {
```

Comment:
Could we use `const&` here?
```suggestion
                        const std::vector<T> &vec) {
```